lib/middleman-navtree/helpers.rb

```diff
@@ -11,12 +11,21 @@ def tree_to_html(value, depth = Float::INFINITY, key = nil, level = 0)
           # This is a file.
           # Get the Sitemap resource for this file.
           # note: sitemap.extensionless_path converts the path to its 'post-build' extension.
-          this_resource = sitemap.find_resource_by_path(sitemap.extensionless_path(value))
-          # Define string for active states.
-          active = this_resource == current_page ? 'active' : ''
-          title = discover_title(this_resource)
-          link = link_to(title, this_resource)
-          html << "<li class='child #{active}'>#{link}</li>"
+          
+          # Make sure the extension path ends with .html (in case we're parsing someting like .adoc)
+          extensionlessPath = sitemap.extensionless_path(value)
+          unless extensionlessPath.end_with? ".html"
+           extensionlessPath << ".html"
+          end
+          
+          this_resource = sitemap.find_resource_by_path(extensionlessPath)
+          if this_resource
```

Comment:
Yes I did. I discovered this when I added ".adoc" to the allowed extensions but didn't have the logic in place to add ".html" if it was missing. In that case it would look for a resource and couldn't find it.
That's why I added this. Otherwise it would fail somewhere down the line with a pretty nonsensical exception.